create-and-save-map/src/main/java/com/esri/arcgismaps/sample/createandsavemap/screens/MainScreen.kt

```diff
@@ -0,0 +1,462 @@
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.layers.Layer
+import com.arcgismaps.portal.PortalFolder
+import com.arcgismaps.toolkit.geoviewcompose.MapView
+import com.esri.arcgismaps.sample.createandsavemap.components.MapViewModel
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialog
+import com.esri.arcgismaps.sample.sampleslib.components.SampleTopAppBar
+
+/**
+ * Main screen layout for the sample app
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun MainScreen(sampleName: String) {
+    // create a ViewModel to handle MapView interactions
+    val mapViewModel: MapViewModel = viewModel()
+
+    val composableScope = rememberCoroutineScope()
+    val snackbarHostState = remember { SnackbarHostState() }
+
+    Scaffold(
+        snackbarHost = { SnackbarHost(hostState = snackbarHostState) },
+        topBar = { SampleTopAppBar(title = sampleName) },
+        content = {
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(it),
+                contentAlignment = Alignment.Center
+            ) {
+                var mapScale by remember { mutableIntStateOf(0) }
+                val map = mapViewModel.arcGISMap
+
+                // map view not shown until the portal is loaded and a basemap has been set
+                if (map != null) {
+                    MapView(
+                        modifier = Modifier
+                            .fillMaxSize(),
+                        arcGISMap = map,
+                        // update the map scale in the UI on map scale change
+                        onMapScaleChanged = { currentMapScale ->
+                            if (!currentMapScale.isNaN()) {
+                                mapScale = currentMapScale.roundToInt()
+                            }
+                        },
+                    )
+                }
+
+                val controlsBottomSheetState =
+                    rememberModalBottomSheetState(skipPartiallyExpanded = true)
+                // show the "Edit map" button only when the bottom sheet is not visible
+                if (!controlsBottomSheetState.isVisible) {
+                    ExtendedFloatingActionButton(
+                        modifier = Modifier
+                            .align(Alignment.BottomEnd)
+                            .padding(bottom = 36.dp, end = 24.dp),
+                        onClick = {
+                            composableScope.launch {
+                                controlsBottomSheetState.show()
+                            }
```

Comment:
Using this may skip the `show` animation of the BottomSheet since it would be an immediate action, I would suggest creating a property to track and set the visibility of the bottomsheet: 
```kotlin
// .... other remember vals
var isBottomSheetVisible by remember { mutableStateOf(false) }

```
```suggestion
                            isBottomSheetVisible = true
```